frontend/javascripts/dashboard/transfer_task_modal.tsx

```diff
@@ -11,67 +12,52 @@ type Props = {
   onCancel: (...args: Array<any>) => any;
   isOpen: boolean;
 };
-type State = {
-  currentUserIdValue: string;
-};
 
-class TransferTaskModal extends React.PureComponent<Props, State> {
-  state: State = {
-    currentUserIdValue: "",
-  };
+const TransferTaskModal: React.FC<Props> = ({ isOpen, onCancel, annotationId, onChange }) => {
+  const [currentUserIdValue, setCurrentUserIdValue] = useState("");
 
-  async transfer() {
-    const annotationId = this.props.annotationId;
+  const handleSelectChange = useCallback((userId: string) => {
+    setCurrentUserIdValue(userId);
+  }, []);
 
+  const transfer = useCallback(async () => {
     if (!annotationId) {
       throw new Error("No annotation id provided");
     }
 
     try {
-      const updatedAnnotation = await transferTask(annotationId, this.state.currentUserIdValue);
-      this.props.onChange(updatedAnnotation);
+      const updatedAnnotation = await transferTask(annotationId, currentUserIdValue);
+      onChange(updatedAnnotation);
     } catch (error) {
       handleGenericError(error as Error);
     }
-  }
+  }, [annotationId, currentUserIdValue, onChange]);
 
-  handleSelectChange = (userId: string) => {
-    this.setState({
-      currentUserIdValue: userId,
-    });
-  };
-
-  render() {
-    if (!this.props.isOpen) {
-      return null;
-    }
+  if (!isOpen) {
+    return null;
+  }
 
-    return (
-      <Modal
-        title="Transfer a Task"
-        open={this.props.isOpen}
-        onCancel={this.props.onCancel}
-        footer={
-          <div>
-            <Button
-              type="primary"
-              onClick={() => this.transfer()}
-              disabled={this.state.currentUserIdValue === ""}
-            >
-              Transfer
-            </Button>
-            <Button onClick={() => this.props.onCancel()}>Close</Button>
-          </div>
-        }
-      >
-        <div className="control-group">
-          <div className="form-group">
-            <UserSelectionComponent handleSelection={this.handleSelectChange} />
-          </div>
+  return (
+    <Modal
+      title="Transfer a Task"
+      open={isOpen}
+      onCancel={onCancel}
+      footer={
+        <div>
+          <Button type="primary" onClick={transfer} disabled={currentUserIdValue === ""}>
+            Transfer
+          </Button>
+          <Button onClick={onCancel}>Close</Button>
```

Comment:
I'd add a small horizontal space between the buttons as this file is already being edited. Currently, it looks like this:
![image](https://github.com/user-attachments/assets/1c37c9a1-d85a-4fcf-9b5d-ced27c831561)
Which can be improved imo